src/reanimated2/js-reanimated/index.ts

```diff
@@ -47,14 +49,21 @@ export const _updatePropsJS = (
   }
 };
 
-const setNativeProps = (
+const updatePropsDOM = (
   component: JSReanimatedComponent,
   style: StyleProps
 ): void => {
   const previousStyle = component.previousStyle ? component.previousStyle : {};
   const currentStyle = { ...previousStyle, ...style };
   component.previousStyle = currentStyle;
-  component.setNativeProps({ style: currentStyle });
+
+  const domStyle = createReactDOMStyle(currentStyle);
+  if (domStyle.transform) {
+    domStyle.transform = createTransformValue(domStyle.transform);
+  }
+  for (const key in domStyle) {
```

Comment:
I think this approach will result in duplicate `transform` object being passed to `domStyle`, you will receive a [warning log](https://github.com/necolas/react-native-web/blob/39b94b194594adebdfd5d4a5b873ced8d387734e/packages/react-native-web/src/exports/StyleSheet/preprocess.js#L203) even if it is working well.

Reply:
@alantoa Hey, what exactly do you mean by "duplicated `transform` object"?

I've compared 551f37ac44e0c20f8f255a95b3bc179084494346 to your suggestion from https://github.com/software-mansion/react-native-reanimated/pull/4352#discussion_r1182346178 and it looks like in both cases we call `createReactDOMStyle` first, then at some point we call `createTransformValue` for `domStyle.transform` and finally we assign its result to `component.style.transform` (and we assign to `domStyle.transform` only once). What's exactly wrong with overwriting `domStyle.transform`, maybe I'm missing something?

Reply:
oh yeah, i misread, never mind! thank you!

Reply:
no problem, thanks!